### PR TITLE
fix: Stop scaling instance labels

### PR DESCRIPTION
**Pull request by kof**
Closes https://github.com/webstudio-is/webstudio/issues/1990

## Description

We were scaling instance labels along with instaces

## Steps for reproduction

1. select instance
2. resize canvas to scale it up and down
3. labels get smaller, but should stay same size while instance scales
4. outline position should stay correct

## Code Review

- [ ] hi @TrySound , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/builder/features/workspace/canvas-tools/outline/use-outline.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { useStore } from "@nanostores/react";
+import type { Instance } from "@webstudio-is/sdk";
+import type { Atom } from "nanostores";
+import { scaleStore } from "~/builder/shared/nano-states";
+
+export const useOutline = (
```
Comment (TrySound) [inline apps/builder/app/builder/features/workspace/canvas-tools/outline/use-outline.ts:6]: Can be computed

Comment (TrySound) [reply apps/builder/app/builder/features/workspace/canvas-tools/outline/use-outline.ts:6]: Ah, got it. You pass different stores with same shape.

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/builder/features/workspace/canvas-tools/outline/use-outline.ts (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/workspace/canvas-tools/outline/use-outline.ts:6]: yes

**Comment by kof** [comment]
Found a bug with scaling during d&d, fixing ...

**Review by GoharAkram — APPROVED** [review]
- [ ] Tested by adding components and scaling the canvas up and down
Works fine now and does not decrease instance label scaling below 100%

**Comment by kof** [comment]
@GoharAkram additionally you needed to test if adding instance outlines the target instance correctly along with placement indicator

**Review by TrySound — APPROVED** [review]

**Comment by TrySound** [comment]
Getting shifted outline
image

**Comment by kof** [comment]
@TrySound good catch
@GoharAkram missed it

File: apps/builder/app/builder/features/workspace/canvas-tools/outline/label.tsx [hunk 2]
```diff
@@ -16,24 +16,22 @@ type LabelRefCallback = (element: HTMLElement | null) => void;
...
     (element: null | HTMLElement) => {
-      if (element === null || instanceRect === undefined) {
+      if (element === null || rect === undefined) {
         return;
       }
       const labelRect = element.getBoundingClientRect();
       let nextPosition: LabelPosition = "top";
-      if (labelRect.height > instanceRect.top) {
-        nextPosition = instanceRect.height < 250 ? "bottom" : "inside";
+      if (labelRect.height > rect.top) {
```
Comment (istarkov) [inline apps/builder/app/builder/features/workspace/canvas-tools/outline/label.tsx:29]: Why we check `height` here? And not the similar dimension like bottom or top?

**Review by istarkov — APPROVED** [review]

File: apps/builder/app/builder/features/workspace/canvas-tools/outline/label.tsx (hunk 2 above)
Comment (kof) [reply apps/builder/app/builder/features/workspace/canvas-tools/outline/label.tsx]: - didn't change this logic and actually forgot to revert the change I made in renaming
- tested logic works correctly, the change you are proposing breaks a use case

**Comment by GoharAkram** [comment]
Hey @kof This is just a final review of the initial PR or has something specifically changed since the last review?

**Review by GoharAkram — APPROVED** [review]
- [ ] This was tested. 

Added instances and checked that scaling canvas above or below 100% doesn't affect the labels. It only scales the component instances.

also the selected instance's outline is restricted to canvas now, This fixes the earlier issue

**Pull request merged**